### ISSUE TITLE
Better information regarding using Person object without ID

### ISSUE THREAD

**Issue by beeradmoore** [rollbar/Rollbar.NET#366]
When attempting to log an error while also adding the Person object the system will timeout.

```
var rollbarConfig = new RollbarConfig("...");
var person = new Rollbar.DTOs.Person();
//var person = new Rollbar.DTOs.Person("someid");
person.Email = "brad@test.com";
rollbarConfig.Person = person;
RollbarLocator.RollbarInstance.Configure(rollbarConfig);
RollbarLocator.RollbarInstance.AsBlockingLogger(TimeSpan.FromSeconds(15)).Log(ErrorLevel.Error, "Test error.");
```

The above code will throw a TimeoutException (Posting a payload to the Rollbar API Service timed-out) if I use the first Person constructor, it will send correctly if I use the second. 

I expected either to be able to log without an ID, to throw an exception, or a console output warning saying that it is not configured correctly. 

It makes perfect sense that ID would need to be defined, and I could also most probably use email as ID if needed. The failure should explain why it failed at least rather than seeming to be a random network issue.

This is using Rollbar.NET 3.0.3 in a dotnetcore 2.1 ASP.NET project running locally on macOS.

**Comment by akornich** [comment]
@beeradmoore , correct Person.Id is a required property, email can be used as an ID. If you tap into RollbarQueuesController.InternalEvent you would be able to see internal SDK errors with more details about what possibly happened that caused a delivery timeout (like, network issue, payload format, missing critical fields in the payload, Rollbar API Service errors, invalid payload, etc.)

**Comment by akornich** [comment]
@beeradmoore , here are general categories of internal SDK errors that we recognize currently: [https://github.com/rollbar/Rollbar.NET/blob/master/Rollbar/InternalRollbarError.cs](https://github.com/rollbar/Rollbar.NET/blob/master/Rollbar/InternalRollbarError.cs)

**Comment by beeradmoore** [comment]
It was a System.TimeoutException with no InnerException set. Would I have to pull from github and add it as a solution to my project?

As a dev implementing Rollbar into an existing service it had wasted my time looking to see why it timed out. 

Is why I was wondering if there is a way to more easily expose this incorrect setup to someone using the nuget package. Maybe even removing the default empty constructor of Person?

**Comment by akornich** [comment]
@beeradmoore , the default empty constructor (as well as the setter for the ID property) was added relatively easily since we have users that would like to be able to assign/re-assign the ID later in the game.  
I'll look into the possibility of collecting errors/exceptions along the processing pipeline per payload so that they could be added as inner exceptions for the timeout exception.

**Comment by beeradmoore** [comment]
It makes perfect sense to update the ID later down in their own cod, just as it makes perfect sense that ID is a required field. If people wanted to update the ID later down in the code, they could just as easily construct a new Person object. That is unless people are creating so many Person object so quickly that there is actually performance improvements of re-using objects. 

Using inner execution would be very useful, just something to indicate that it is a required field. If I get time on the weekend I'll see if I can do a PR for it.

**Comment by akornich** [comment]
#370

**Comment by akornich** [comment]
this item is already well addressed by #370 and #379.
